### PR TITLE
Dynamic NFT support

### PR DESCRIPTION
**Pull request by tifrel**
This PR:

- Introduces missing fields for handling the new v2 minting paradigm and dynamic NFTs
  - Improves royalty handling
  - Makes `u64`/`U64` handling consistent with events emitted by the smart contracts
- Reworks the construction of payloads for downstream services by removing useless constructors

Checklist:

- [x] does a DB schema change
  - [x] hasura permissions are updated
  - [ ] the indexer is halted
  - [x] the corresponding minterop-producer PR is ready
  - [x] the corresponding minterop-consumer PR is ready
- [x] changes pubsub payloads
  - [ ] the indexer is halted
  - [x] the corresponding minterop-producer PR is ready
  - [x] the corresponding minterop-consumer PR is ready
- [ ] deprecates hasura permissions
  - [ ] frontend won't break
- [ ] creates hasura triggers/actions/events etc.
  - [ ] postgres DB user has been granted appropriate permissions

### CODE REVIEW

**Review by sainthiago — APPROVED** [review]

**Comment by tifrel** [comment]
CI fails because hasura inconsistencies arise before the actual DB was migrated. This is to be expected. Created a task to fix the CI here.

File: migrations/2024-02-29-130650_complete-metadata-creation/up.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+alter table nft_metadata
```
Comment (bh2smith) [inline migrations/2024-02-29-130650_complete-metadata-creation/up.sql:1]: Are any of these planned to be not null?

File: src/db_rows.rs [hunk 2]
```diff
@@ -91,6 +91,11 @@ pub struct NftMetadata {
...
     pub price: Option<BigDecimal>,
     pub minters_allowlist: Option<Vec<String>>,
+    pub royalties: Option<serde_json::Value>,
+    pub royalty_percent: Option<BigDecimal>,
+    pub max_supply: Option<BigDecimal>,
+    pub last_possible_mint: Option<NaiveDateTime>,
+    pub is_locked: Option<bool>,
 }
 
 impl NftMetadata {
```
Comment (bh2smith) [inline src/db_rows.rs:101]: Isn't this about the same as derive::Default?

In or you might be able to remove several lines of code here with `..default::Default()` (at least for all the Options).

Unless we are trying to be explicit.

File: src/rpc_payloads.rs [hunk 3]
```diff
@@ -37,59 +44,6 @@ pub enum RpcMessage {
     },
 }
 
-impl RpcMessage {
```
Comment (bh2smith) [inline src/rpc_payloads.rs:40]: I guess none of these functions was used anywhere...

**Review by bh2smith — APPROVED** [review]
Looks good to me. Nice clean changes!

File: src/rpc_payloads.rs (hunk 3 above)
Comment (tifrel) [reply src/rpc_payloads.rs:40]: They were used, but it was kinda stupid. At first they were doing some conversions like `AccountId` to `String`, but especially with the metadata they just became a pass-through that was just really unnecessary. You will see it on the corresponding indexer PR.

File: migrations/2024-02-29-130650_complete-metadata-creation/up.sql (hunk 1 above)
Comment (tifrel) [reply migrations/2024-02-29-130650_complete-metadata-creation/up.sql:1]: Yes, mb_internal_id should not be null for Mintbase v2 metadata, all the others are still optional but might not be null.

File: src/db_rows.rs (hunk 2 above)
Comment (tifrel) [reply src/db_rows.rs]: Right, gonna fix

**Pull request merged**